File: Ex30.cpp
/* Código tomado de mi repositorio, ejercicio 29.

#include <iostream>
#include <cmath>


    double L=1.0;
    double t=1.0;
    
    double maxk=190;
    double dt=t/maxk;
    double n=30;
    double dx=L/n;
    double b=pow(dx,2)/2;


int main()
{

    for (double k=0.0; k<maxk;k++)
    {
        for (double i=0.0; i<n;i++)
        {
			
            double res1=4*pow(sin(k*dx/2),2)*i;
            std::cout<<k<<" "<<res1<<std::endl;
        }
        
    }
    return 0;
}

File: Ex30.py
#Código tomado de mi repositorio, ejercicio 29

import os
import numpy as np
import matplotlib.pyplot as plt

data = np.loadtxt("Ex29.dat")
plt.figure(figsize=(20,5))
plt.subplot(1,2,1)
plt.imshow(data/np.max(data[:,0]))
plt.colorbar()
plt.subplot(1,2,2)
plt.plot(data[:,0], data[:,1]/np.max(data[:,1]))
plt.title("Diffusive equation")
plt.xlabel("t")
plt.ylabel("Diffusion")
plt.savefig("diff.png")
plt.show()
